packages/salesforcedx-vscode-core/src/commands/forceDebuggerStop.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import {
+  ContinueResponse,
+  EmptyParametersGatherer,
+  ParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+interface QueryResponse {
+  status: number;
+  result: QueryResult;
+}
+
+interface QueryResult {
+  size: number;
+  totalSize: number;
+  records: QueryRecord[];
+}
+
+interface QueryRecord {
+  Id: string;
+}
+
+export type IdSelection = { id: string };
+export class IdGatherer implements ParametersGatherer<IdSelection> {
+  private readonly sessionIdToUpdate: string;
+
+  public constructor(sessionIdToUpdate: string) {
+    this.sessionIdToUpdate = sessionIdToUpdate;
+  }
+
+  public async gather(): Promise<ContinueResponse<IdSelection>> {
+    return { type: 'CONTINUE', data: { id: this.sessionIdToUpdate } };
+  }
+}
+
+export class DebuggerSessionDetachExecutor extends SfdxCommandletExecutor<
+  IdSelection
+> {
+  public build(data: IdSelection): Command {
+    return new SfdxCommandBuilder()
+      .withArg('force:data:record:update')
+      .withFlag('--sobjecttype', 'ApexDebuggerSession')
+      .withFlag('--sobjectid', data ? data.id : '')
+      .withFlag('--values', 'Status="Detach"')
```

Comment:
IDE1 updated its status to 'Kill' with a status message that says one of the org's admins killed the session because people were sharing development in a sandbox org. Scratch orgs are supposed to be per developer so I'm just using detach.